endpoint.sh

```diff
@@ -48,37 +42,25 @@ then
     then
         :
     else
-	      sed -i -e "s#${endpoints_pioneer[$wss_pioneer_index]}#${endpoints_pioneer[0]}#g" $build_endpoint
+	      sed -i -e "s#${endpoints[$wss_pioneer_index]}#${endpoints[0]}#g" $build_endpoint
     fi
 else
-    for i_a in ${!endpoints_api[@]}
+    for i_a in ${!endpoints[@]}
     do
-        if [ $i_a == 0 ]
+	    echo $i_a
+        echo ${endpoints[$i_a]}
+        sleep 5s
+        telemetry_height=$(node telemetry-block.js)
+        sleep 5s
+        wss_height=$(node compare-heights.js ${endpoints[$i_a]})
+        let "wss_height += 2"
+        echo $wss_height
+        if [ $wss_height -ge $telemetry_height ]
         then
-            :
+            sed -i -e "s#${endpoints[$wss_pioneer_index]}#${endpoints[$i_a]}#g" $build_endpoint
+            break
         else
-	    echo $i_a
-	    echo ${endpoints_api[$i_a - 1]}
-            echo ${endpoints_api[$i_a]}
-            sed -i -e "s#'${endpoints_api[$i_a - 1]}'#'${endpoints_api[$i_a]}'#g" ~/webui/node_modules/@polkadot/rpc-provider/defaults.js
-            sleep 5s
-            telemetry_height=$(node telemetry-block.js)
-	          sleep 5s
-            wss_api_index="$i_a"
-            wss_height=$(node compare-heights.js)
-            let "wss_height += 2"
-	          echo $wss_height
-            if [ $wss_height -ge $telemetry_height ]
-            then
-                wss_pioneer="${endpoints_pioneer[$i_a]}"
-                wss_pioneer_index="$i_a"
-                sed -i -e "s#${endpoints_pioneer[$i_a - 1]}#${endpoints_pioneer[$i_a]}#g" $build_endpoint
```

Comment:
I think this was originally incorrect? 
`sed -i -e "s#${endpoints_pioneer[$i_a - 1]}#${endpoints_pioneer[$i_a]}#g" $build_endpoint`
should have been:
`sed -i -e "s#${endpoints_pioneer[$wss_pioneer_index]}#${endpoints_pioneer[$i_a]}#g" $build_endpoint`